### PR TITLE
Remove `Send` bound

### PR DESCRIPTION
**Pull request by hannobraun**
As far as I can tell, the `Send` bound doesn't serve any purpose and removing it didn't seem to have any adverse effects.

### CODE REVIEW

**Comment by ArtemGr** [comment]
What if I'm sharing the terminal between threads? I happen to be doing just that.

**Comment by hannobraun** [comment]
@ArtemGr I believe your use case will not be affected at all. `Send` is automatically derived by the Rust compiler whenever that is possible. That means, `TerminfoTerminal<T>` will be `Send`, if `T` is `Send`.

Currently, `T` is explicitly required to be `Send`, even though `TerminfoTerminal` itself doesn't actually need that capability. That means, you can never use a `T` that is not `Send`, even if you don't actually need `TerminfoTerminal<T>` to be `Send`.

So, to summarize: Unless I misunderstand the subtleties of `Send`, this change should make my use case possible, without changing anything about yours :)

**Comment by ArtemGr** [comment]
Oh, cool. Maybe we're talking about the differend `Send`s even! I'm only sharing the https://docs.rs/term/0.4.4/term/type.StdoutTerminal.html between threads. Thanks for clearing things out, though.

**Comment by ArtemGr** [comment]
I wonder if it even worth it to reuse the terminal. Right now all it saves is one `Box::new` it seems. On the other hand, if `fn stdout()` will make some extra checks in the future, such as `isatty` as discussed in #54, then suddenly I'm saving on I/O.

**Comment by Stebalien** [comment]
Sorry for the long delay, LGTM

**Comment by hannobraun** [comment]
No problem. Thanks for looking into it!
